xml/System.IO/Path.xml

```diff
@@ -1457,46 +1489,58 @@ Unlike the string overload, this method doesn't normalize directory separators.
-- "\\\ComputerName\SharedFolder" (a UNC path).  
-  
-- "\\\\\?\C:" (a DOS device path, supported in .NET Core 1.1 and later versions and in .NET Framework 4.6.2 and later versions)
-   
- For more information on file paths on Windows, see [File path formats on Windows systems](~/docs/standard/io/file-path-formats.md). For a list of common I/O tasks, see [Common I/O Tasks](~/docs/standard/io/common-i-o-tasks.md).  
-  
-## Examples  
- The following example demonstrates a use of the `GetPathRoot` method.  
-  
- [!code-cpp[System.IO.Path Members#8](~/samples/snippets/cpp/VS_Snippets_CLR_System/system.IO.Path Members/CPP/pathmembers.cpp#8)]
- [!code-csharp[System.IO.Path Members#8](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.IO.Path Members/CS/pathmembers.cs#8)]
- [!code-vb[System.IO.Path Members#8](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.IO.Path Members/VB/pathmembers.vb#8)]  
-  
- ]]></format>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This method does not verify that the path or file name exists.
+
+This method will normalize directory separators.
+
+A string is "effectively empty" if:
+
+- In Windows, calling `IsEmpty` on this string returns `true`, or all its characters are spaces (' ').
+- In Unix, calling <xref:System.String.IsNullOrEmpty> on this string returns `true`.
+
+Possible patterns for the string returned by this method are as follows:
+
+- `null` (`path` was null or an empty string).
+
+- An empty string (`path` specified a relative path on the current drive or volume).
+
+- "\" (Unix: `path` specified an absolute path on the current drive).
+
+- "X:" (Windows: `path` specified a relative path on a drive, where X represents a drive or volume letter).
+
+- "X:\" (Windows: `path` specified an absolute path on a given drive).
+
+- "\\\ComputerName\SharedFolder" (Windows: a UNC path).
+
+- "\\\\\?\C:" (Windows: a DOS device path, supported in .NET Core 1.1 and later versions and in .NET Framework 4.6.2 and later versions)
+ 
+For more information on file paths on Windows, see [File path formats on Windows systems](~/docs/standard/io/file-path-formats.md). For a list of common I/O tasks, see [Common I/O Tasks](~/docs/standard/io/common-i-o-tasks.md).
+
+## Examples
+The following example demonstrates a use of the `GetPathRoot` method.
+
+[!code-cpp[System.IO.Path Members#8](~/samples/snippets/cpp/VS_Snippets_CLR_System/system.IO.Path Members/CPP/pathmembers.cpp#8)]
+[!code-csharp[System.IO.Path Members#8](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.IO.Path Members/CS/pathmembers.cs#8)]
+[!code-vb[System.IO.Path Members#8](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.IO.Path Members/VB/pathmembers.vb#8)]
+
+          ]]></format>
         </remarks>
-        <exception cref="T:System.ArgumentException">
-          <paramref name="path" /> contains one or more of the invalid characters defined in <see cref="M:System.IO.Path.GetInvalidPathChars" />.  
```

Comment:
Does the method not throw if `path` contains invalid characters?

Reply:
That's what I discovered on .NET Core 3.0. Let me test this again but on Framework, I suspect this comment applies only for that version.

Reply:
@rpetrusha confirmed:
- In 3.0, we don't throw when we pass "`\0`", `null` or `string.Empty`.
    - If we pass `null` or `string.Empty`, `GetPathRoot` returns `null`.
    - If we pass "`\0`", `GetPathRoot` returns `string.Empty`.
- In Framework, we throw if we pass "`\0`" or `string.Empty`. We don't throw if we pass `null`.

@JeremyKuhne is this a breaking change? Should I open an issue to bring back the exception in .NET Core for those cases?

Reply:
@rpetrusha I brought back the exception, but made sure to mention it is only thrown in .NET Framework (for both root causes).

Reply:
It was a deliberate breaking change in 2.1. Partially to align with the Span overloads (there are no null Spans), partially to align better with the other APIs and user expectations.